test/test_register.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+from taskgraph.transforms.task import payload_builders
+
+from mozilla_taskgraph import register
+
+
+def test_payload_builders(graph_config):
+    expected_payload_builders = {
+        "scriptworker-shipit",
+    }
+    register(graph_config)
+    missing = expected_payload_builders - set(payload_builders)
+    if missing:
+        print("The following expected payload builders are missing:")
+        print("  " + "\n  ".join(sorted(missing)))
+    assert not missing
+
+
+@pytest.mark.parametrize(
+    "extra_config,raises",
+    (
+        pytest.param({}, False, id="no_extra_config"),
+        pytest.param(
+            {"shipit": {"product": "foo"}},
+            False,
+            id="shipit_valid",
+        ),
+        pytest.param(
+            {"shipit": {"product": "foo", "extra": "bar"}},
+            True,
+            id="shipit_invalid",
+        ),
+    ),
+)
+def test_graph_config(make_graph_config, extra_config, raises):
+    # Shipit config is valid
+    graph_config = make_graph_config(extra_config=extra_config)
+    try:
+        register(graph_config)
+        assert raises is False
+    except Exception:
+        assert raises is True
```

Comment:
Nit: pytest provides the context manager `raises()`[1] which makes this piece smaller to write 😃 

```suggestion
    with expectation:
        register(graph_config)
```

`expectation` can be parametrized with either `pytest.raises(Exception)` or a null context manager. More info there[2]

[1] https://docs.pytest.org/en/7.4.x/reference/reference.html#pytest.raises
[2] https://docs.pytest.org/en/7.4.x/example/parametrize.html#parametrizing-conditional-raising

Reply:
Neat, TIL!